sdk/identity/azure-identity/src/samples/Azure Cache For Redis/Jedis/Azure-AAD-Authentication-With-Jedis.md

```diff
@@ -0,0 +1,199 @@
+        } catch (JedisException e) {
+            // Handle The Exception as required in your application.
+            e.printStackTrace();
+
+            // Check if the client is broken, if it is then close and recreate it to create a new healthy connection.
+            if (jedis.isBroken() || accessToken.isExpired()) {
+                jedis.close();
+                jedis = createJedisClient(cacheHostname, 6380,"USERNAME", getAccessToken(clientCertificateCredential, trc), useSsl);
+            }
+        }
+
+        // Close the Jedis Client
+        jedis.close();
+    }
+
+    private static Jedis createJedisClient(String cacheHostname, int port, String username, AccessToken accessToken, boolean useSsl) {
+        return new Jedis(cacheHostname, port, DefaultJedisClientConfig.builder()
+                .password(accessToken.getToken())
+                .user(username)
+                .ssl(useSsl)
+                .build());
+    }
+
+    private static AccessToken getAccessToken(TokenCredential tokenCredential, TokenRequestContext trc) {
+        return tokenCredential.getToken(trc).block();
+    }
+```
+
+#### Authenticate with AAD Azure Jedis Wrapper
+This sample is intended to assist in the migration from Jedis to `AzureJedisClientBuilder`. It focuses on side-by-side comparisons for similar operations between the two libraries.
+
+Familiarity with the Jedis and Azure Identity client libraries is assumed. If you're new to the Azure Identity library for Java, see the docs for [Azure Identity](https://docs.microsoft.com/azure/developer/java/sdk/identity) and [Jedis](https://www.javadoc.io/doc/redis.clients/jedis/latest/index.html) rather than this guide.
+
+##### Migration benefits
+
+A natural question to ask when considering whether or not to adopt a new version or library is what the benefits of doing so would be. Jedis by itself doesn't support Azure AD authentication with token generation, failure retries, broken connection handling, and cache reauthentication. Using `AzureJedisClient` will improve developer productivity and code maintainability.
+
+##### Client instantiation
+
+In Jedis, you create a `Jedis` object via a public constructor. The constructor accepts the cache host name and port number. It authenticates using the access keys. For example:
+
+```java
+import redis.clients.jedis.Jedis;
+
+Jedis jedis = new Jedis("<host name>", <port number>);
+jedis.auth("<username>", "<token>");
+jedis.set("key", "value");
+jedis.close();
+```
+
+With `AzureJedisClient`, client instances are created via builders. The builder accepts the:
+
+- Cache host name
+- Port number to connect to
+- Username set on the cache
+- Optional retry options to configure retry
+- Token credential object that's used to generate a token
+
+See the following example of setting up the Azure Jedis client.
+
+![image](https://user-images.githubusercontent.com/5430778/166531908-3ed78774-3672-4bf8-a6cf-9c6ef6c0bdff.png)
```

Comment:
Convert this image to a Markdown table. This makes it easy to update down the road.